### ISSUE TITLE
Performance issues with large number of subscriptions

### ISSUE THREAD

**Issue by thitemple** [elmish/elmish#151]
### Description

I'm subscribing to a GraphQL data source and generating a subscription with that. I'm receiving about 10 to 12 messages every second, that is making the application very laggy and slow.
I'm using Elmish with React Native. 

I tried different things to understand the problem, the only thing I was able to do to resolve the issue was to encapsulate the following piece of code in a ReactComponent and manage the data in its state. So, eliminating Elmish from the scenario. I didn't want to do that, because this data will be used elsewhere.

### Repro code
Here's the relevant code:

```let subscribeQuery = @"
        subscription onTrackerUpdated {
            trackerUpdated {
                id
                latitude
                longitude
            }
        }
    "
    let queryObservable = client.subscribe<TrackerUpdateResponse> { query = (gql subscribeQuery) }
    let subscription (dispatch : Msg -> unit) =
        queryObservable.subscribe (TrackerUpdated >> dispatch)
    Cmd.batch [queryCmd; Cmd.ofSub subscription]
```

### Expected and actual results

Please provide the expected and actual results.

### Related information 

* elmish version: 1.0.3
* fable-compiler version: 1.3.17
* fable-core version: 1.3.17
* Operating system:  Windows 10

**Comment by MangelMaxime** [comment]
@thitemple Do you have any debugger tools activated ?

Like:
- `Program.withDebugger`
- `Program.withConsoleTrace`

Just to be sure, you only subscribe once to the query and stop listening to unuseful query ?

**Comment by Zaid-Ajaj** [comment]
@thitemple You can introduce a small delay (maybe 500 ms) between each received message before dispatching it, I haven't tested this though but you can try it
```fs
let throttle dispatch delay = MailboxProcessor.Start <| fun inbox -> 
    let rec loop() = async {
        let! msg = inbox.Receive()
        do! Async.Sleep delay 
        dispatch msg 
        do! loop() 
    }

    loop() 

let subscription (dispatch : Msg -> unit) =
       // dispatch only in 500 ms intervals 
       let throttledDispatch = throttle dispatch 500
       queryObservable.subscribe (TrackerUpdated  >> throttledDispatch )
```

**Comment by MangelMaxime** [comment]
Hum, I am receiving several message per seconds in my App and and it's working fine. Mine is slow because I have a list of thousand elements that is being updated so it take times. :p

Also a quick thing to note, when using HMR it's possible that even if you think you register only once to an Event it's note the case in dev mode.
For example, the [browser navigation](https://github.com/elmish/browser/blob/v1.x/src/navigation.fs#L78) module is attaching it's event each time the application is updated. For this one it's not a problem because there is a check in the code to only send the message in the Elmish app if the URL changed.

This is something I discovered when working with [Thoth.Toast](https://github.com/MangelMaxime/Thoth/blob/master/src/Thoth.Elmish.Toast/Toast.fs#L406-L424).

For your problem, in order to test what is the source of the problem you could try to use a Minimal Elmish application. Like just receiving the messages and displaying a counter in the view. 
This can help you isolate the problem, because in theory Elmish has been made for working with intensive events. And perhaps you also need to use `lazyViews` in order to not re-render everything on each message but only the part that has been changed.

**Comment by thitemple** [comment]
So, the view in case has only a map and a list, they are both displaying information related to the data received by the subscription. The map displays the elements according to the coordinates and the list displays other information like (is it on time or not). That is to say that the view is simple and using `lazyView` wouldn't have much effect in this case because everything in the view depends on this data.

I wrapped only the map in a `ReactComponent` class and that fixed the issue, but the list is not being updated anymore. I could probably put the list in there as well.

To have a better idea, I have 10 elements being tracked in the map and the list, each of this elements trigger an update every half second, so I have 10 updates in less than a second every second.

I have the data stored in an array, so I can remove an element that needs to be updated from the array and then I can add it once again. 

I'm pretty sure that I only subscribed to it once because I added some console.logs to track it.

The other workaround that fixed the issue for me (for now at least) was instead of subscribing to the changes, I'm pulling the data for the 10 elements at once every 2000ms using setInterval. Doing this everything is smooth.
